builder/dockerfile/builder.go

```diff
@@ -264,7 +264,8 @@ func (b *Builder) build(stdout io.Writer, stderr io.Writer, out io.Writer) (stri
 	total := len(b.dockerfile.Children)
 	for _, n := range b.dockerfile.Children {
 		if err := b.checkDispatch(n, false); err != nil {
-			return "", err
+			// Specify in which line the parsing failed
+			return "", fmt.Errorf("Line %d: %s", n.StartLine, err)
```

Comment:
I think we should specifically say `Dockerfile parse error line %d`... so the error is self contained, as there could be other types of Dockerfile error in future.